azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -969,223 +930,336 @@ def do_retry():
 class ReconnectState(object):
     """
     Class which holds reconnect states as class variables.  Created to make code that reads like an enum without using an enum.
-
-    WAITING_TO_RECONNECT: This stage is in a waiting period before reconnecting.  This state implies
-    that the user wants the pipeline to be connected.  ie. After a successful connection, the
-    state will change to LOGICALLY_CONNECTED
-
-    LOGICALLY_CONNECTED: The client wants the pipeline to be connected.  This state is independent
-    of the actual connection state since the pipeline could be logically connected but
-    physically disconnected (this is a temporary condition though.  If we're logically connected
-    and physically disconnected, then we should be waiting to reconnect.
-
-    LOGICALLY_DISCONNECTED: The client does not want the pipeline to be connected or the pipeline had
-    a permanent errors error and was forced to disconnect.  If the state is LOGICALLY_DISCONNECTED, then the pipeline
-    should be physically disconnected since there is no reason to leave the pipeline connected in this state.
     """
 
-    WAITING_TO_RECONNECT = "WAITING_TO_RECONNECT"
-    LOGICALLY_CONNECTED = "LOGICALLY_CONNECTED"
-    LOGICALLY_DISCONNECTED = "LOGICALLY_DISCONNECTED"
+    CONNECTED = "CONNECTED"  # Client is connected (as far as it knows)
+    DISCONNECTED = "DISCONNECTED"  # Client is disconnected
+    CONNECTING = "CONNECTING"  # Client is in the process of connecting
+    DISCONNECTING = "DISCONNECTING"  # Client is in the process of disconnecting
+    REAUTHORIZING = "REAUTHORIZING"  # Client is in the process of reauthorizing (disconn->conn)
```

Comment:
>(disconn->conn) [](http://example.com/codeflow?start=81&length=15)

what does disconn->conn mean in this context?  disconnected->connected?  disconnected->connecting? etc.

Reply:
This is what I get for trying to keep the comment under 100 characters. I'll make this clearer.